src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -369,6 +371,29 @@ private static void AddActionDataToBuilder(
 
         builder.Metadata.Add(action);
 
+        if (metadataCountAfterPopulating != metadataCountBeforePopulating)
```

Comment:
I'm a little wary of a change that specifically targets `IProducesResponseTypeMetadata` and not the `IEndpointMetadataProvider` interface in general -- especially because this will be brittle to any changes in how we represent response metadata in the future.

Have you investigated what it would look like to fix this by applying `IEndpointMetadataProvider`?

Reply:
I took advantage of the fact that endpointMetadataPopulator.PopulatesMetadata ends up calling this code https://github.com/dotnet/aspnetcore/blob/main/src%2FHttp%2FHttp.Results%2Fsrc%2FOkOfT.cs#L67 which is the one that reports the type and status code of the response